pkg/index/job/correction/service/corrector.go

```diff
@@ -70,24 +69,46 @@ type correct struct {
 	checkedIndexCount         atomic.Uint64
 	correctedOldIndexCount    atomic.Uint64
 	correctedReplicationCount atomic.Uint64
+
+	indexReplica               int
+	streamListConcurrency      int
+	bboltAsyncWriteConcurrency int
 }
 
 const filemode = 0o600
 
-func New(cfg *config.Data, discoverer discoverer.Client) (Corrector, error) {
-	d := file.Join(os.TempDir(), "bbolt")
-	file.MkdirAll(d, os.ModePerm)
-	dbfile := file.Join(d, "checkedid.db")
-	bolt, err := bbolt.New(dbfile, "", os.FileMode(filemode))
-	if err != nil {
+func New(opts ...Option) (_ Corrector, err error) {
+	c := new(correct)
+	for _, opt := range append(defaultOpts, opts...) {
+		if err := opt(c); err != nil {
+			oerr := errors.ErrOptionFailed(err, reflect.ValueOf(opt))
+			e := &errors.ErrCriticalOption{}
+			if errors.As(oerr, &e) {
+				log.Error(err)
+				return nil, oerr
+			}
+			log.Warn(oerr)
+		}
+	}
+	if err := c.kvsInit(); err != nil {
 		return nil, err
 	}
+	return c, nil
+}
+
+func (c *correct) kvsInit() error {
```

Comment:
> **NOTE**
To make it easier to implement tests in the future,  I have separated initialize logic of bbolt into method 🙏